src/main/scala/org/broadinstitute/hail/driver/LinearMixedModelCommand.scala

```diff
@@ -0,0 +1,167 @@
+import org.broadinstitute.hail.stats.LinearMixedModel
+import org.broadinstitute.hail.utils._
+import org.kohsuke.args4j.{Option => Args4jOption}
+
+object LinearMixedModelCommand extends Command {
+
+  def name = "lmmreg"
+
+  def description = "Use a kinship-based linear mixed model to estimate the genetic component of phenotypic variance (narrow-sense heritability) and optionally test each variant for association"
+
+  class Options extends BaseOptions {
+    @Args4jOption(required = true, name = "-y", aliases = Array("--phenotype"), usage = "Response sample annotation")
+    var ySA: String = _
+
+    @Args4jOption(name = "-c", aliases = Array("--covariates"), usage = "Covariate sample annotations, comma separated")
+    var covSA: String = ""
+
+    @Args4jOption(required = true, name = "-k", aliases = Array("--kinshipfilter"), usage = "Variant filter expression for the kinship matrix")
+    var kinshipFiltExprVA: String = _
+
+    @Args4jOption(name = "-a", aliases = Array("--assocfilter"), usage = "Variant filter expression for association testing")
+    var assocFiltExprVA: String = _
+
+    @Args4jOption(name = "-n", aliases = Array("--noassoc"), usage = "Only fit the global model, no association testing")
+    var noAssoc: Boolean = false
+
+    @Args4jOption(name = "--useml", usage = "Use ML instead of REML throughout")
+    var useML: Boolean = false
+
+    @Args4jOption(name = "--globalroot", usage = "Global annotation root, a period-delimited path starting with `global'")
+    var rootGA: String = "global.lmmreg"
+
+    @Args4jOption(name = "--varoot", usage = "Variant annotation root, a period-delimited path starting with `va'")
+    var rootVA: String = "va.lmmreg"
+
+    @Args4jOption(name = "--delta", usage = "Fixed delta value to use in the global model, overrides fitting delta")
+    var delta: java.lang.Double = _
+
+    @Args4jOption(hidden = true, name = "--sparsitythreshold", usage = "AF threshold above which to use dense genotype vectors in rotation")
+    var sparsityThreshold: Double = 1d
+
+    @Args4jOption(hidden = true, name = "--forceblock", usage = "Force using Spark's BlockMatrix to compute kinship")
+    var forceBlock: Boolean = false
+
+    @Args4jOption(hidden = true, name = "--forcegrammian", usage = "Force using Spark's RowMatrix.computeGrammian to compute kinship")
+    var forceGrammian: Boolean = false
+  }
+
+  def newOptions = new Options
+
+  def supportsMultiallelic = false
+
+  def requiresVDS = true
+
+  def run(state: State, options: Options): State = {
+    val vds = state.vds
+
+    val pathVA = Parser.parseAnnotationRoot(options.rootVA, Annotation.VARIANT_HEAD)
+    val pathGA = Parser.parseAnnotationRoot(options.rootGA, Annotation.GLOBAL_HEAD) // FIXME: included only to check for errors
+
+    val symTab = Map(
+      "s" -> (0, TSample),
+      "sa" -> (1, vds.saSignature))
+
+    val ec = EvalContext(symTab)
+
+    def toDouble(t: BaseType, code: String): Any => Double = t match {
```

Comment:
The Boolean to Double conversion is defined in `RichBoolean` in  `org.broadinstitute.hail.utils.richUtils`.  That said, in a latter pass, I intend to refactor the extraction of covariates to a common class for all regressions implemented so far.